src/main/java/com/dnastack/ga4gh/dataconnect/controller/DataConnectController.java

```diff
@@ -49,6 +49,18 @@ public TableData search(@RequestBody DataConnectRequest dataConnectRequest,
             log.debug("Request: /search query= {}", dataConnectRequest.getSqlQuery());
             tableData = trinoDataConnectAdapter
                 .search(dataConnectRequest.getSqlQuery(), request, parseCredentialsHeader(clientSuppliedCredentials), null);
+            // Motivation for the following while loop is to resolve auth errors in Trino on the POST request rather than during subsequent GET requests.
+            // see https://github.com/DNAstack/data-connect-trino/pull/45
+            // Using following while loop could potentially pose an issue in situations of high activity within Trino, where job's time in queue
```

Comment:
My only comment was going to be this timeout issue, which you've already thought about. I'd suggest addressing this sooner rather than later. This is the kind of problem that will be completely fine until a sufficient load happens to cause a run-away feedback loop — many concurrent queries timing out, leading to many retries, leading to even more queries exectuing/queued before they get killed by the idle query killing job.

I'd strongly recommend proactively adding a time limit to this code, after which you could fallback to returning the most recent queued page URL. Ideally the limit would be based on some empirical data for how long queries take to go from queued to executing, but barring that you could also make it slightly less than the standard HTTP response timeout used in our microservice HTTP clients, ingress controllers, etc.

Reply:
I agree. If possible, we should add the timeout to this loop sooner. By addressing the timeout issue, we also ensure that there will be little to no chance where:
* a request-handling thread is being blocked (due to this loop) for a long time,
* and the blocking loop may reduce the service responsiveness, i.e., a number of requests per second.